Page exo avec parsage

christen14 opened the pull request:
Page web d'un exercice codé en dur avec parsage.
Bug avec le tag latex : \qquad.

Commit:
Page exo avec parsage

Page web d'un exercice codé en dur avec parsage.
Bug avec le tag latex : \qquad.

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -82,102 +64,145 @@ <h4>Correction : </h4>
-        \finexercice`;
+\correction
+$\rho =\sqrt{4+2\sqrt2}$, $\theta ={3\pi \over8}$; \qquad 
+$\rho =4$, $\theta =-{\pi \over 10}$; \qquad
```
pour le \qquad, je propose d'ajouter, dans la fonction qui remplace des choses dans les chaînes (les \\ , les accents etc), une ligne qui remplace tout simplement \qquad par un espace simple " ", ou éventuellement un quadruple espace html "&emsp;" (l'espace simple c'est &nbsp;" (pour "non-breaking space")

Commit:
fetch de l'exercice 23

Petits changements dans exercice-avec-parsage.
Mathjax exercice-fetch-23 ne load pas correctement.

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -2,7 +2,7 @@
 <html>
```
Ok, très bien, on y est presque, sur cette page.
Le mathjax n'a pas l'air de bien marcher : peut-être que lorsque du contenu dynamique arrive, il faut dire à Mathjax de relancer le rendu. Je sais que dans mathjax2 il était possible de faire ça, il faut se documenter pour voir comme faire ceci en mathjax3

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -28,24 +28,6 @@
-
     <script>
       // Remplace tous les  \'[lettre]  par la lettre avec accent
       function replaceAccents(chaine) {
```
ici, recoder cette fonction, par exemple comme ce qu'a suggéré chatGPT (dans le chat teams)

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -82,101 +64,145 @@ <h4>Correction : </h4>
-        \end{align*}
-        \fincorrection
-        \finexercice`;
+\video{8nXKgqMsucU}
```
rajouter une fonction qui vire ces commandes, genre \video{...}, ou alors qui les remplace par u lien html vers la vidéo youtube. Mais dans tous les cas, faire une fonction qui cleane ça. On peut commencer par juste virer ces commandes, on verra après si on remplace par un lien

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -82,101 +64,145 @@ <h4>Correction : </h4>
-    // enfin, remplacement des accents.
+    
+    // Mots clés devant figurant dans chaque exercice 
+    var motsObligatoires= [String.raw`\exercice`, String.raw`\finexercice`, String.raw`\enonce`,String.raw`\finenonce`];
```
là il faut virer tous ces String.raw, le code devient illisible. Surtout pour de petites chaînes. Remplacer par : 
const motsObligatoires = ["\\exercice","\\finexercice" etc.

Le String.raw est utile pour ne pas écrire à la main plein de double slashs, mais pour des petites chaines le prix à payer est trop grand, ça rend le code trop lourd.

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -82,101 +64,145 @@ <h4>Correction : </h4>
+    
+    //Si l'exercice est correctement écrit :
+
+    // Gestion de bug avec le tag LaTeX : \begin{enumerate}
```
ok, mais il faut aussi gérer les \begin{itemize}\end{itemize} qui marchent pareil (listes non numérotées)

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -82,101 +64,145 @@ <h4>Correction : </h4>
+    var beginEnumerate = monExo.indexOf(String.raw`\begin{enumerate}`);
+    // Suppression du tag s'il est présent dans l'exercice
+    if (beginEnumerate != -1){ 
+      monExo = monExo.replaceAll(String.raw`\begin{enumerate}`, '');
```
tant qu'à faire, il faudrait remplacer par des éléments html en fait : on remplace les \begin{enumerate} par des <ul>, les items par des <li>.  Ou alors on fait lowtech mais il faut au moins remplacer les item par un tiret, et aller à la ligne après l'item

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -82,101 +64,145 @@ <h4>Correction : </h4>
-
-
+    // Affichage de la page
+    htmlText =
```
houlà, ça il faut éviter. Il faut la structure html en dur dans la page, et avec des divs ou des span qui ont des 'id', et on modifie ça. Mais on ne modifie pas tout comme ça, car ça rend le code difficilement lisible. Il est préférable que le template html soit lisible tel quel, dans la partie html.

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -2,7 +2,7 @@
 <html>
```
i y a des remaarques en commun avec le document précédent je crois

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -82,102 +68,147 @@ <h4>Correction : </h4>
+      date = entete[2];
+
+      // Récupération de vidéo
+      const debutVideo = exo.indexOf(String.raw`\video`);
```
en fait là je pense qu'il faudrait une expression régulière pour traiter ça, genre une expression régulière qui remplace \\video{*} par un espace

Commit:
Merge remote-tracking branch 'upstream'

Commit:
Ajout de exercice-fetch-23

Commit:
Merge remote-tracking branch 'upstream'

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
           return chaine;
       } 
 
+    let exo23 = 'https://raw.githubusercontent.com/exo7math/exercices-exo7/master/exercices/ex000023.txt';
```
changer nom de variable : mettre 'URLexo23'

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
-        \finexercice`;
-
-
+fetch(exo23)
```
si possible, préférer syntaxe avec await, mais ceci est non prioritaire

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
+    var beginEnumerate = fetchedText.indexOf(String.raw`\begin{enumerate}`);
+    // Suppression du tag s'il est présent dans l'exercice
+    if (beginEnumerate != -1){ 
+      fetchedText = fetchedText.replaceAll(String.raw`\begin{enumerate}`, '');
```
virer les String.raw

dmegy replied:
il ne faut pas remplacer par "", il faut remplacer par "\n<br>" pour aller à la laigne

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
+    // Suppression du tag s'il est présent dans l'exercice
+    if (beginEnumerate != -1){ 
+      fetchedText = fetchedText.replaceAll(String.raw`\begin{enumerate}`, '');
+      fetchedText = fetchedText.replaceAll(String.raw`\item`, '');
```
remplacer plutôt par "-&emsp;"

dmegy replied:
en fait , remplacer plutôt les \item par "<br> -&emsp;"
(on va à l aligne d'abord)

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
+    if (beginEnumerate != -1){ 
+      fetchedText = fetchedText.replaceAll(String.raw`\begin{enumerate}`, '');
+      fetchedText = fetchedText.replaceAll(String.raw`\item`, '');
+      fetchedText = fetchedText.replaceAll(String.raw`\end{enumerate}`, '');
```
remplacer le \end{enumerate} par un "<br>"

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
+    const debutVideo = fetchedText.indexOf(String.raw`\video`);
+    if (debutVideo != -1){ // S'il y a une vidéo
+      let video = document.getElementById('h1');
+      video.textContent = fetchedText.slice(debutVideo, debutVideo+100);
```
remplacer ça par une expression régulière du genre du genre \video{*}

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
+    exercice.textContent = fetchedText;
+
+    // Mots clés devant figurant dans chaque exercice 
+    var motsObligatoires = [String.raw`\exercice`, String.raw`\finexercice`, String.raw`\enonce`,String.raw`\finenonce`];
```
pareil, enlever les String.raw

dmegy commented on affichage_exercices/exercice-avec-parsage.html:
```diff
@@ -2,7 +2,7 @@
   <head>
     <meta charset="UTF-8">
-    <title>Page sans fetch sans parsage</title>
+    <title>exo20/8 avec parsage</title>
```
commentaire général : commentez le code

dmegy replied:
éventuellement, améliorer les noms de variables  lorsque possible

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
+    entete = entete.split(", ");
+    
+    numExercice.textContent = "Exercice " + entete[0];
+    auteur.textContent = "Auteur : " + entete[1];
```
ici, ne pas faire comme ça : utiliser une balise <span> avec un id pour juste modifier l'auteur, pas tout le texte.

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -81,103 +83,106 @@ <h4>Correction : </h4>
+    var entete = fetchedText.slice(debutEntete+1, finEntete);
+    entete = entete.split(", ");
+    
+    numExercice.textContent = "Exercice " + entete[0];
```
innerHTML plutôt que textContent ?

dmegy commented on affichage_exercices/exercice-fetch-23.html:
```diff
@@ -29,24 +29,26 @@
-      <!-- Entete -->
-      <h1 id="num"></h1>
+      <h1 id="numExercice"></h1>
       <h3 id="auteur"><h3>
```
ne pas faire ça : faire
< h3> Auteur : < span id="auteur">? < /span>< /h3>

Ceci permet que les choix de formatage, les choix de forme, ne soient pas dans le code javascript, mais dans la partie html.

Commit:
Modifications dans les 2 fichiers

Suppression des String.raw``
Affichage avec DOM pour exercice-avec-parsage.html
Modification de la fonction de remplacement de symboles spéciaux
Implémentation du lien pour video
